spec.bs

```diff
@@ -718,50 +779,37 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
-           |trustedBiddingSignals|, |browserSignals| », and |perBuyerTimeout|.
-        1. Let |duration| be the current time minus |startTime| in milliseconds.
-        1. If |generatedBidResult| is an [=abrupt completion=], [=iteration/continue=].
-        1. Let |generatedBidIDL| be the result of [=converted to an IDL value|converting=]
-          |generatedBidResult| to a {{GenerateBidOutput}}.
-        1. If an exception was caught in the previous step, [=iteration/continue=].
-        1. Let |groupHasAdComponents| be true.
-        1. If |interestGroup|'s [=interest group/ad components=] is null:
-          1. Set |groupHasAdComponents| be false.
-        1. Let |generatedBid| be the result of [=converting GenerateBidOutput to generated bid=]
-          with |generatedBidIDL|, |ig|, |isComponentAuction|, and |groupHasAdComponents|.
-        1. Set |generatedBid|'s [=generated bid/bid duration=] to |duration|.
+        1. Let |generatedBid| be the result of [=generate a bid=] given |isComponentAuction|,
+          |allTrustedBiddingSignals|, |auctionSignals|, a [=map/clone=] of |browserSignals|,
+          |perBuyerSignals|, |perBuyerTimeout|, and |ig|.
         1. If |generatedBid| is failure, [=iteration/continue=].
-        1. Set |generatedBid|'s [=generated bid/interest group=] to |ig|.
+        1. If [=query generated bid k-anonymity count=] given |generatedBid| returns false:
+          Note: [=Generate a bid=] is now rerun with only k-anonymous [=interest group/ads=] to give
+          the buyer a chance to [=generate a bid=] for k-anonymous [=interest group/ads=]. Allowing
+          the buyer to first [=generate a bid=] for non-k-anonymous [=interest group/ads=] provides a
+          mechanism to bootstrap the k-anonymity count, otherwise no [=interest group/ads=] would
+          ever trigger [=increment k-anonymity count=] and all ads would fail
+          [=query k-anonymity count=].
+          1. Let |originalAds| be |ig|'s [=interest group/ads=].
+          1. Let |originalAdComponents| be |ig|'s [=interest group/ad components=].
+          1. Set |ig|'s [=interest group/ads=] to a new empty [=list=] of
+            [=interest group ad=].
+          1. Set |ig|'s [=interest group/ad components=] to a new [=list/is empty|empty=]
+            [=list=] of [=interest group ad=].
+          1. [=list/For each=] |ad| in |originalAds|:
+            1. If [=query ad k-anonymity count=] given |ig| and |ad|'s
+              [=interest group ad/render url=] returns true, [=list/append=] |ad| to |ig|'s
+              [=interest group/ads=].
+          1. [=list/For each=] |adComponent| in |originalAdComponents|:
+            1. If [=query component ad k-anonymity count=] given |adComponent|'s
+              [=interest group ad/render url=] returns true, [=list/append=] |adComponent| to |ig|'s
+              [=interest group/ad components=].
+          1. Let |generatedBid| be the result of [=generate a bid=] given |isComponentAuction|,

```

Comment:
```suggestion
          1. Set |generatedBid| to the result of [=generate a bid=] given |isComponentAuction|,
```

Since it's already initialized above.